install_config/monitoring/configuring-etcd-monitoring.adoc

```diff
@@ -0,0 +1,188 @@
+prometheus-operator   37m
+----
++
+It might take up to a minute for the `etcd` service monitor to start.
+
+. Now you can navigate to the Web interface to see more information about status of `etcd` monitoring. To get the URL, run:
++
+[subs="quotes"]
+----
+$ *oc -n openshift-monitoring get routes*
+NAME                HOST/PORT                                                                           PATH      SERVICES            PORT      TERMINATION   WILDCARD
+...
+prometheus-k8s      prometheus-k8s-openshift-monitoring.apps.msvistun.origin-gce.dev.openshift.com                prometheus-k8s      web       reencrypt     None
+----
++
+Using `https`, navigate to the URL listed for `prometheus-k8s`. Log in.
+
+. Ensure the user belongs to the `cluster-monitoring-view` role. This role provides access to viewing cluster monitoring UIs. For example, to add user `developer` to `cluster-monitoring-view`, run:
+
+  $ oc adm policy add-cluster-role-to-user cluster-monitoring-view developer
++
+
+. In the Web interface, log in as the user belonging to the `cluster-monitoring-view` role.
+
+. Click `Status`, then `Targets`.
+
+. You should see the `etcd` entry. This means that `etcd` is being monitored.
++
+image::etcd-no-certificate.png[]
+
+However, Prometheus is not yet able to authenticate against `etcd`, and so cannot gather metrics. Follow these steps to configure Prometheus authentication against `etcd`:
+
+. Copy the `/etc/etcd/ca/ca.crt` and `/etc/etcd/ca/ca.key` credentials files from the master node to the local machine:
++
+[subs="quotes"]
+----
+$ ssh -i gcp-dev/ssh-privatekey cloud-user@35.237.54.213
+...
+----
+
+. Create the `openssl.cnf` file with these contents:
++
+----
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, keyEncipherment, digitalSignature
+extendedKeyUsage=serverAuth, clientAuth
+----
+
+. Generate the `etcd.key` private key file:
++
+[subs="quotes"]
+----
+$ openssl genrsa -out etcd.key 2048
+----
+
+. Generate the `etcd.csr` certificate signing request file:
++
+[subs="quotes"]
+----
+$ openssl req -new -key etcd.key -out etcd.csr -subj "/CN=etcd" -config openssl.cnf
+----
+
+. Generate the `etcd.crt` certificate file:
++
+[subs="quotes"]
+----
+$ openssl x509 -req -in etcd.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out etcd.crt -days 365 -extensions v3_req -extfile openssl.cnf
+----
+
+. Put the credentials into format used by OpenShift:
++
+----
+cat <<-EOF > etcd-cert-secret.yaml
+apiVersion: v1
+data:
+  etcd-client-ca.crt: "$(cat ca.crt | base64 --wrap=0)"
+  etcd-client.crt: "$(cat etcd.crt | base64 --wrap=0)"
+  etcd-client.key: "$(cat etcd.key | base64 --wrap=0)"
+kind: Secret
+metadata:
+  name: kube-etcd-client-certs
+  namespace: openshift-monitoring
+type: Opaque
+EOF
+----
++
+This creates file `etcd-cert-secret.yaml`.
+
+. Apply the credentials file to the cluster:
+
```

Comment:
It is better to change to
Copy the file etcd-cert-secret.yaml to master and apply the credentials file to the cluster

Then user know where to apply the file

Reply:
You don't need to copy etcd-cert-secret.yaml, you apply it from the local computer. Not changing the instruction.